conan/external_tools/vswhere_test.py

```diff
@@ -0,0 +1,169 @@
+import os
+import unittest
+
+import nose
+
+from conans import tools
+from conans.errors import ConanException
+from conans.model.version import Version
+from conans import __version__ as client_version
+from conans.model import settings
+
+
+class vswhereTest(unittest.TestCase):
+
+    # Environment supossed:
+    #   - BuildTools 14 (2015)
+    #   - VS Community 14 (2015)
+    #
+    #   - BuildTools 15 (2017) OR VS Community 15 (2017)
+    modern_products = 1 # 2017 or higher versions -> vswhere()
+    all_modern_products = 2 # 2017 or higher versions -> vswhere(products=["*"])
+    modern_and_legacy_products = 2 # 2017 and lower versions -> vswhere(legacy=True)
```

Comment:
why modern + legacy are 2 but modern = 2 and legacy = 1?

Reply:
I will update the comments and you will understand